### PR TITLE
Document fedimintd/src/fedimintd.rs

### PR DESCRIPTION
**Pull request by isaack-njama**
References #3807

### CODE REVIEW

File: fedimintd/src/fedimintd.rs [hunk 1]
```diff
@@ -32,9 +33,11 @@ use crate::attach_default_module_init_params;
 /// Time we will wait before forcefully shutting down tasks
 const SHUTDOWN_TIMEOUT: Duration = Duration::from_secs(10);
 
+/// Default meta value to use during config generation
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:36]: It's the name of a environent variable that allows passing extra metadata to pass to dkg.

File: fedimintd/src/fedimintd.rs [hunk 2]
```diff
@@ -32,9 +33,11 @@ use crate::attach_default_module_init_params;
...
 pub const FM_EXTRA_DKG_META_VAR: &str = "FM_EXTRA_DKG_META";
 
 #[derive(Parser)]
+/// Struct to define server options
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:40]: Style: "Struct" here is redundant, the user can see it's a `struct` both in code and rendered documentation. Just "Server options" is more succinct.

File: fedimintd/src/fedimintd.rs [hunk 3]
```diff
@@ -128,8 +131,11 @@ fn parse_map(s: &str) -> anyhow::Result<BTreeMap<String, String>> {
 /// }
 /// ```
 pub struct Fedimintd {
+    /// Registry for initializing server modules
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:134]: "Registry of available module kinds". 

Module Init is kind of a constructor for a module (Fedimint could possibly construct multiple module of the same kind) - with bunch of function to handle module before it's really initialized.

File: fedimintd/src/fedimintd.rs [hunk 4]
```diff
@@ -128,8 +131,11 @@ fn parse_map(s: &str) -> anyhow::Result<BTreeMap<String, String>> {
...
     pub server_gens: ServerModuleInitRegistry,
+    /// Registry for configuring server module generation parameters
     pub server_gen_params: ServerModuleConfigGenParamsRegistry,
+    /// Represents version hash of Fedimintd
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:138]: It doesn't "represent". It is a fedimint's git sha1 of the source code commit used to build the binary.

File: fedimintd/src/fedimintd.rs [hunk 5]
```diff
@@ -182,13 +188,13 @@ impl Fedimintd {
...
     }
-
+    /// Attach default server modules to Fedimintd instance
     pub fn with_default_modules(self) -> Self {
         self.with_module(LightningInit)
             .with_module(MintInit)
             .with_module(WalletInit)
     }
-
+    /// Run a Fedimintd server asynchronously
```
Comment (dpc) [inline fedimintd/src/fedimintd.rs:197]: It's not asynchronous. This call will block forever and run the server.

**Review by dpc — DISMISSED** [review]

**Review by elsirion — DISMISSED** [review]
Looks good now, but two process remarks:
* If you don't understand the thing you are about to document either put in the work to understand it or don't document it, half the comments being wrong isn't good use of reviewer's time
* Commits shouldn't undo changes from previous commits in a PR, there's `git squash`, `rebase -i`, `commit --amend` for that.

**Comment by isaack-njama** [comment]
> Looks good now, but two process remarks:
> 
> * If you don't understand the thing you are about to document either put in the work to understand it or don't document it, half the comments being wrong isn't good use of reviewer's time
> * Commits shouldn't undo changes from previous commits in a PR, there's `git squash`, `rebase -i`, `commit --amend` for that.

Noted

File: fedimintd/src/fedimintd.rs [hunk 6]
```diff
@@ -183,13 +189,13 @@ impl Fedimintd {
...
     }
-
+    /// Attach default server modules to Fedimintd instance
     pub fn with_default_modules(self) -> Self {
         self.with_module(LightningInit)
             .with_module(MintInit)
             .with_module(WalletInit)
     }
-
+    /// Block thread and run a Fedimintd server
```
Comment (elsirion) [inline fedimintd/src/fedimintd.rs:198]: Please don't remove newlines between function definitions.

**Review by elsirion — APPROVED** [review]

**Review by justinmoon — APPROVED** [review]

**Pull request merged**